engine/api/src/main/java/io/deephaven/engine/table/iterators/ObjectColumnIterator.java

```diff
@@ -0,0 +1,95 @@
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ChunkSource;
+import io.deephaven.engine.table.Table;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * {@link ColumnIterator} implementation for {@link ChunkSource chunk sources} of objects.
+ */
+public final class ObjectColumnIterator<TYPE> extends ColumnIterator<TYPE, ObjectChunk<TYPE, ? extends Any>> {
+
+    /**
+     * Create a new ObjectColumnIterator.
+     * 
+     * @param chunkSource The {@link ChunkSource} to fetch values from; must have {@link ChunkSource#getChunkType() *
+     *        chunk type} of {@link ChunkType#Object}
+     * @param rowSequence The {@link RowSequence} to iterate over
+     * @param chunkSize The internal buffer size to use when fetching data
+     */
+    public ObjectColumnIterator(
+            @NotNull final ChunkSource<? extends Any> chunkSource,
+            @NotNull final RowSequence rowSequence,
+            final int chunkSize) {
+        super(validateChunkType(chunkSource, ChunkType.Object), rowSequence, chunkSize);
+    }
+
+    /**
+     * Create a new ObjectColumnIterator.
+     *
+     * @param chunkSource The {@link ChunkSource} to fetch values from; must have {@link ChunkSource#getChunkType() *
+     *        chunk type} of {@link ChunkType#Object}
+     * @param rowSequence The {@link RowSequence} to iterate over
+     */
+    public ObjectColumnIterator(
+            @NotNull final ChunkSource<? extends Any> chunkSource,
+            @NotNull final RowSequence rowSequence) {
+        this(validateChunkType(chunkSource, ChunkType.Object), rowSequence, DEFAULT_CHUNK_SIZE);
+    }
+
+    /**
+     * Create a new ObjectColumnIterator.
+     *
+     * @param table {@link Table} to create the iterator from
+     * @param columnName Column name for iteration; must have {@link ChunkSource#getChunkType() chunk type} of
+     *        {@link ChunkType#Object}
+     */
+    public ObjectColumnIterator(@NotNull final Table table, @NotNull final String columnName) {
+        this(table.getColumnSource(columnName), table.getRowSet(), DEFAULT_CHUNK_SIZE);
+    }
+
+    @Override
+    ObjectChunk<TYPE, ? extends Any> castChunk(@NotNull final Chunk<? extends Any> chunk) {
+        return chunk.asObjectChunk().asTypedObjectChunk();
+    }
+
+    @Override
+    public TYPE next() {
+        maybeAdvance();
+        return currentData.get(currentOffset++);
+    }
+
+    @Override
+    public void forEachRemaining(@NotNull final Consumer<? super TYPE> action) {
+        consumeRemainingByChunks(() -> {
+            final int currentSize = currentData.size();
+            while (currentOffset < currentSize) {
+                action.accept(currentData.get(currentOffset++));
+            }
+        });
+    }
+
+    public Stream<TYPE> stream() {
```

Comment:
1. Because I didn't need those, and `ObjectColumnIterator` is not replicated.
2. Because I didn't find the right tool for the job the first time I looked, and I got a little lazy since I didn't need it.
Either way, I'm adding those 3 now.